core/agent/agent.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package agent
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/juju/errors"
+	"github.com/juju/utils/shell"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/juju/osenv"
+	"github.com/juju/juju/juju/paths"
+	"github.com/juju/juju/service/common"
+)
+
+const (
+	maxAgentFiles = 20000
+
+	agentServiceTimeout = 300 // 5 minutes
+)
+
+// AgentConf returns the data that defines an init service config
+// for the identified agent.
+func AgentConf(info AgentInfo, renderer shell.Renderer) common.Conf {
+	conf := common.Conf{
+		Desc:          fmt.Sprintf("juju agent for %s", info.Name),
+		ExecStart:     info.Cmd(renderer),
+		Logfile:       info.LogFile(renderer),
+		Env:           osenv.FeatureFlags(),
+		Timeout:       agentServiceTimeout,
+		ServiceBinary: info.Jujud(renderer),
+		ServiceArgs:   info.ExecArgs(renderer),
+	}
+
+	switch info.Kind {
+	case AgentKindMachine:
+		conf.Limit = map[string]int{
+			"nofile": maxAgentFiles,
+		}
+	case AgentKindUnit:
+		conf.Desc = "juju unit agent for " + info.ID
+	}
+
+	return conf
+}
+
+//find all the agents available in the machine
```

Comment:
Function comments in Go start with the function name,
// FindAgents finds all the ..........